test/e2e/storage/sanity.go

```diff
@@ -996,3 +997,21 @@ func WaitForPodsWithLabelRunningReady(c clientset.Interface, ns string, label la
 		})
 	return pods, err
 }
+
+func checkForLeftoverVolumes(numWorkers int) {
+	var cmd string
+	switch os.Getenv("TEST_DEVICEMODE") {
+	case "direct":
+		cmd = "sudo ndctl list"
+	case "lvm":
+		cmd = "sudo lvs --foreign --noheadings"
+	}
+	for worker := 1; worker <= numWorkers; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, cmd)
+		out, _ := ssh.CombinedOutput()
+		// TODO: remove this debug print before merging:
+		framework.Logf("checkforleftover-DEBUG(%d): cmd:[%s] out:[%s]", worker, cmd, out)
+		Expect(string(out)).To(BeEmpty())
```

Comment:
This introduces the new requirement that the cluster must be completely unused (for example, none of the example apps running), otherwise testing will fail. That's okay for the CI, but may be too limited for testing on a development system.

I'm also worried that the commands (`ndctl list` and `lvs`) will report things that should be ignored on a more complex cluster.

We can either make this check optional and only activate it in the CI or enhance the check: list volumes before the test runs and again afterwards, then expect the results to be the same.

Reply:
> list volumes before the test runs and again afterwards

That is good method IMO. 
It also makes the failure point standing out better. Right now, we get series of failures, only 1st being significant and all remaining are flagged only because the list cmd keeps indicating that one leftover entry. By recording state before and after test, we get rid of any previous state impact. And instead of 29 failures, we will see 1.

Reply:
I should mention another drawback, though: we cannot run tests in parallel anymore. We don't do that anyway at the moment because some tests are disruptive and would interfere with other tests, but with volume checking we are forced to run sequentially also for other tests. I'm fine with that.